src/kontrol/foundry.py

```diff
@@ -947,8 +956,12 @@ def check_cells_equal(cell: str, nodes: Iterable[KCFG.Node]) -> bool:
     nodes = [apr_proof.kcfg.node(int(node_id)) for node_id in node_ids]
     check_cells = ['K_CELL', 'PROGRAM_CELL', 'PC_CELL', 'CALLDEPTH_CELL']
     check_cells_ne = [check_cell for check_cell in check_cells if not check_cells_equal(check_cell, nodes)]
+
     if check_cells_ne:
-        raise ValueError(f'Nodes {node_ids} cannot be merged because they differ in: {check_cells_ne}')
+        if check_same_loop(nodes):
+            pass
+        else:
+            raise ValueError(f'Nodes {node_ids} cannot be merged because they differ in: {check_cells_ne}')
```

Comment:
```suggestion
        if not check_same_loop(nodes):
            raise ValueError(f'Nodes {node_ids} cannot be merged because they differ in: {check_cells_ne}')
```

Reply:
We already check above that `len(nodes) >= 2`, so this more compact inline check should work.